src/platform/bouffalolab/BL602/DiagnosticDataProviderImpl.cpp

```diff
@@ -248,36 +249,24 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetWiFiBssId(ByteSpan & BssId)
 
 CHIP_ERROR DiagnosticDataProviderImpl::GetWiFiSecurityType(app::Clusters::WiFiNetworkDiagnostics::SecurityTypeEnum & securityType)
 {
-    using app::Clusters::WiFiNetworkDiagnostics::SecurityTypeEnum;
-    securityType = SecurityTypeEnum::kUnspecified;
-    // int authmode;
-
-    // authmode     = mgmr_get_security_type();
-    // securityType = MapAuthModeToSecurityType(authmode);
+    securityType = (app::Clusters::WiFiNetworkDiagnostics::SecurityTypeEnum) wifi_mgmr_get_security_type();
     return CHIP_NO_ERROR;
```

Comment:
This should be returning error if we're not connected to wifi.

Reply:
Fixed